niceman/distributions/singularity.py

```diff
@@ -95,14 +100,30 @@ def identify_distributions(self, files):
 
         images = []
         remaining_files = set()
+        url = None
 
         for file_path in files:
             try:
-                image = json.loads(self._session.execute_command(['singularity',
-                    'inspect', file_path])[0])
+                if file_path.startswith('shub:/'):
+                    # Correct file path for path normalization in retrace.py
+                    if not file_path.startswith('shub://'):
+                        file_path = file_path.replace('shub:/', 'shub://')
+                    temp_path = "{}.simg".format(uuid.uuid4())
+                    with chpwd(tempfile.gettempdir()):
```

Comment:
I think there should be some log message (lgr.info) to announce that we will download the thing to be "traced"